### PR TITLE
New test development - CPU-OFFLINE-STORAGE-WORKLOAD

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
Defined a new test in xml file, CPU-OFFLINE-STORAGE-WORKLOAD.

1. Added steps for disk addition in Azure VM, and workload script in the PS1 script body. 
2. Relocated the VM restart step to the end.
3. Added steps for disk mounting in Bash script.
4. Adjust the run-time flow.
5. Changed fio command with different job numbers due to CPU state change bug.
6. Increased the wait-time from 15 to 30 seconds of Run-LinuxCmd interval timer.
7. Added the log print

**TEST RESULT**

[LISAv2 Test Results Summary]
Test Run On           : 07/07/2020 23:30:25
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Override VM size      : Standard_D32as_v4
Initial Kernel Version: 5.3.0-1028-azure
Final Kernel Version  : 5.8.0-rc1+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:25

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-STORAGE-WORKLOAD                                                      PASS                22.54

[LISAv2 Test Results Summary]
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Initial Kernel Version: 5.3.0-1028-azure
Final Kernel Version  : 5.8.0-rc1+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:39

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-STORAGE-WORKLOAD                                                      PASS                35.82

[LISAv2 Test Results Summary]
Test Run On           : 07/07/2020 23:59:41
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Initial Kernel Version: 4.15.0-1089-azure
Final Kernel Version  : 5.8.0-rc1+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:25

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-STORAGE-WORKLOAD                                                      PASS                22.86

### CODE REVIEW

File: Testscripts/Linux/CPUOfflineKernelBuild.sh [hunk 1]
```diff
@@ -22,6 +22,49 @@ function Main() {
...
+			SetTestStateAborted
+			exit 0
+		fi
+
+		chmod 766 /etc/fstab
+
+		echo $ext_uuid $basedir/data ext4 defaults,nofail 1 2 >> /etc/fstab
+		LogMsg "$?: Updated /etc/fstab file with uuid information"
+		ret=$(cat /etc/fstab)
+		LogMsg "$?: Displayed the contents in /etc/fstab"
```
Comment (LiliDeng) [inline Testscripts/Linux/CPUOfflineKernelBuild.sh:67]: did you display $ret?

File: Testscripts/Windows/CPU-Offline.ps1 [hunk 2]
```diff
@@ -28,6 +31,8 @@ function Main {
 
 	try {
 		$testResult = $resultFail
+		$isStorageWorkloadEnable = 0
```
Comment (LiliDeng) [inline Testscripts/Windows/CPU-Offline.ps1:33]: why assign this twice

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/CPU-Offline.ps1 (hunk 2 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/CPU-Offline.ps1:33]: Fixed in 3873e9bd982c480954f2da56d376acfcd68592e0

File: Testscripts/Linux/CPUOfflineKernelBuild.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/CPUOfflineKernelBuild.sh:67]: Fixed in 3873e9bd982c480954f2da56d376acfcd68592e0

**Comment by juhlee-microsoft** [comment]
Working on the test execution on RHEL 8.1

**Comment by juhlee-microsoft** [comment]
**TEST RESULT**

[LISAv2 Test Results Summary]
Test Run On           : 07/08/2020 21:56:55
ARM Image Under Test  : RedHat : RHEL : 8.1 : latest
Initial Kernel Version: 4.18.0-147.5.1.el8_1.x86_64
Final Kernel Version  : 5.8.0-rc1+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:34

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-STORAGE-WORKLOAD                                                      PASS                31.59

**Comment by juhlee-microsoft** [comment]
[LISAv2 Test Results Summary]
Test Run On           : 07/08/2020 23:15:55
ARM Image Under Test  : SUSE : sles-15-sp1 : gen1 : latest
Initial Kernel Version: 4.12.14-8.33-azure
Final Kernel Version  : 5.8.0-rc1-8.33-azure+
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:27

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 CPU-OFFLINE-STORAGE-WORKLOAD                                                      PASS                23.44

**Review by LiliDeng — APPROVED** [review]

**Comment by LiliDeng** [comment]
> [LISAv2 Test Results Summary]
> Test Run On : 07/08/2020 23:15:55
> ARM Image Under Test : SUSE : sles-15-sp1 : gen1 : latest
> Initial Kernel Version: 4.12.14-8.33-azure
> Final Kernel Version : 5.8.0-rc1-8.33-azure+
> Total Test Cases : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
> Total Time (dd:hh:mm) : 0:0:27
> 
> ## ID TestArea TestCaseName TestResult TestDuration(in minutes)
> ```
> 1 CORE                 CPU-OFFLINE-STORAGE-WORKLOAD                                                      PASS                23.44
> ```

I am curious why the final kernel version contains azure+

**Pull request merged**